code/modules/status_system/statusEffects.dm

```diff
@@ -345,209 +345,111 @@
+					. = 0 //normal dose
+				if(0.1 to 1)
+					. = 1 //you might feel sick
+				if(1 to 3)
+					. = 2 //you're getting into dangerous teritory
+				if(3 to 5)
+					. = 3 //you're at a 50/50 of kicking it
+				if(5 to 7)
+					. = 4 //more like 70/30 now
+				if(7 to 10)
+					. = 5 //you will die without treatment
+				if(7 to INFINITY)
+					. = 6 //you will die.
 
 		onUpdate(timePassed)
-
 			var/mob/M = null
 			if(ismob(owner))
 				M = owner
+			else
+				return ..(timePassed)
 
-
-			damage_tox = (sqrt(min(duration, 2 MINUTES)/20 + 5) - 0.5)
-			damage_brute = damage_tox/2
-
-			stage = get_stage(duration)
+			stage = get_stage(M.radiation_dose)
 			switch(stage)
-				if(1)
+				if(0)
 					howMuch = ""
+					damage_tox = 0
+					damage_burn = 0
+				if(1)
+					howMuch = "barely " //you'll be fine
+					damage_tox = prob(10)
+					damage_burn = 0
 				if(2)
-					howMuch = "significantly "
+					howMuch = "slightly " //you don't feel so good
+					damage_tox = prob(25)
+					damage_burn = prob(5)
 				if(3)
-					howMuch = "very much "
+					howMuch = "moderately " // not great, not terrible
+					damage_tox = prob(50)
+					damage_burn = prob(50)
 				if(4)
-					howMuch = "extremely "
+					howMuch = "extremely " //oh no, you're very sick
+					damage_tox = prob(75)
+					damage_burn = prob(60)
 				if(5)
-					howMuch = "horribly "
+					howMuch = "fatally " //congrats, you're dead in a minute
+					damage_tox = 1 + prob(60)
+					damage_burn = 1 + prob(40)
+				if(6)
+					howMuch = "totally " // you are literally dying in seconds
+					damage_tox = rand(2,5)
+					damage_burn = rand(1,4)
+			if(stage > 0)
+				visible = TRUE
+			else
+				visible = FALSE
 
-			if(M)
-				SEND_SIGNAL(M, COMSIG_MOB_GEIGER_TICK, stage)
-				if(locate(/obj/item/implant/health) in M)
-					src.visible = 1
-				else
-					src.visible = 0
-				if (prob(stage - !!(M.traitHolder?.hasTrait("stablegenes"))) && (M.bioHolder && !M.bioHolder.HasEffect("revenant")))
+			if(stage > 0 && !isdead(M))
+				if (prob(max(stage-(2+M.traitHolder?.hasTrait("stablegenes")),0)**2) && (M.bioHolder && !M.bioHolder.HasEffect("revenant")))
 					boutput(M, "<span class='alert'>You mutate!</span>")
 					M.bioHolder.RandomEffect("either")
-				if(!ON_COOLDOWN(M, "n_radiation_stun_check", 1 SECOND) && prob(stage-1) && M.bioHolder && !M.bioHolder.HasEffect("revenant"))
-					M.changeStatus("weakened", 5 SECONDS)
+				if(!ON_COOLDOWN(M, "radiation_stun_check", 1 SECOND) && prob((stage-1)**2) && M.bioHolder && !M.bioHolder.HasEffect("revenant"))
+					M.changeStatus("weakened", 3 SECONDS)
 					boutput(M, "<span class='alert'>You feel weak.</span>")
 					M.emote("collapse")
-			update_lights()
-			icon_state = "nradiation[stage]"
+				if(!ON_COOLDOWN(M, "radiation_vomit_check", 5 SECOND) && prob(stage**2) && M.bioHolder && !M.bioHolder.HasEffect("revenant"))
+					M.changeStatus("weakened", 3 SECONDS)
+					boutput(M, "<span class='alert'>You feel sick.</span>")
+					M.vomit()
```

Comment:
gut feel is that this is probably too much stunning, especially at higher stages, but can't say with certainty

Reply:
It's pretty variable because of the `prob()` but at low stages it's pretty rare, at mid stages it's an irritation, and at end stages it's every few seconds, but at that point you're already well into dying. I can lower it if you prefer, but it feels about right to me.